### PR TITLE
[Validator] fix File constraint tests on 32bit PHP

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | alternative to #52259
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/Validator/Tests/Constraints/FileTest.php [hunk 1]
```diff
@@ -22,11 +22,11 @@ class FileTest extends TestCase
     /**
      * @dataProvider provideValidSizes
      */
-    public function testMaxSize($maxSize, $bytes, $binaryFormat)
+    public function testMaxSize($maxSize, $bytes, $binaryFormat, $exceeds32BitMax)
     {
         $file = new File(['maxSize' => $maxSize]);
 
-        $this->assertSame($bytes, $file->maxSize);
+        $this->assertSame($exceeds32BitMax && 4 === \PHP_INT_MAX ? (string) $bytes : $bytes, $file->maxSize);
```
Comment (stof) [inline src/Symfony/Component/Validator/Tests/Constraints/FileTest.php]: even easier IMO: don't add `$exceeds32BitMax` but update the data provider to change the expected value for `$bytes` based on `\PHP_INT_SIZE`

Comment (stof) [inline src/Symfony/Component/Validator/Tests/Constraints/FileTest.php]: This uses the wrong constant. The max integer is not 4...

**Review by stof — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Validator/Tests/Constraints/FileTest.php (hunk 1 above)
Comment (xabbuh) [reply src/Symfony/Component/Validator/Tests/Constraints/FileTest.php]: good idea

Comment (xabbuh) [reply src/Symfony/Component/Validator/Tests/Constraints/FileTest.php]: oops, indeed

**Review by alexandre-daubois — APPROVED** [review]
Neat! :slightly_smiling_face:

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Comment by xabbuh** [comment]
looks like this made things worse, #52284 fixes it

**Pull request merged**